### ISSUE TITLE
Optimize data Lookup

### ISSUE THREAD

**Issue by Tille12** [Derstefan/IOGame#12]
### Right now
Example Blueprint lookup data (screenshots show implementation of first `Blueprint` subclass `Genesis`):
* one Enum for all holding `EBlueprint` objects, with the variables `id` and `description`. The  `EBlueprint` objecets like `GENESIS` serve as a key for finding the right information within `dataBlueprints`
![image](https://user-images.githubusercontent.com/35866129/102083451-de4fbb00-3e13-11eb-9b6b-c77f0c167a7f.png)

* one abstract class `dataBlueprints`...
...holding configuration of each specific `Blueprint` subclass object (like `Genesis`) within the function `dataBlueprints.setBlueprintConfiguration(EBlueprint blueprint)` and ...
![image](https://user-images.githubusercontent.com/35866129/102083599-1eaf3900-3e14-11eb-9c48-1a2a09784587.png)
...`dataBlueprints.instantiateBlueprint(EBlueprint eBlueprint)`, which holds the specific constructor to the specific sublass of Blueprint, e.g. `Genesis(...)`
![image](https://user-images.githubusercontent.com/35866129/102083689-42727f00-3e14-11eb-8084-5a6a7274d9b3.png)

### Optimization
* Maybe there is a way to put all information from `EBlueprint` and `dataBlueprints` into an `Enum` like `EBlueprint`
--> the `EBlueprint` objects already serve as a key within `dataBlueprints`
* Maybe the system "one specific blueprint - one specific `Blueprint` subclass" is too overkill, and it would serve the purpose well to just create general `Blueprint` objects with the help of the lookup system behind

**Comment by soerenreichardt** [comment]
The idea itself sounds nice, but what do you think about having the constants for the configuration in a file? Maybe `.yaml` format or something. I think many games store their configurations this way.

**Comment by Tille12** [comment]
Yes sure. I am new to this problem and have no experience on the best practice, so I just implemented it like so to make it work. As it does work, I agree that it is not super important right now to change it, but during the process I had to think a lot about how to do this properly, and wanted to collect some ideas on the topic here.
The idea of having simple lookup files, which can be changed without changing any "code" is for sure desirable.
